java/core/src/test/java/com/microsoft/bond/testmodel/MyEnum.java

```diff
@@ -78,6 +78,30 @@ public static MyEnum get(int value) {
         }
     }
 
+    public static MyEnum parse(String str) {
```

Comment:
If the behavior matches the true `Enum`'s `valueOf`, the name should definitely be the same.

Turning a function that could comfortably be 
```
String -> Nullable EnumT
```
into
```
String -> Nullable EnumT, throws IllegalArgException
// we pinky-swear never to return null
```
is a bad Java convention I am broadly willing to violate. For the sake of expediency, I'm okay with this, but I want to revisit it later. Might make sense to do so at the same time as we revisit the top-level APIs.